ocean-client/addons/command-center/src/commands/set-reinvest.ts

```diff
@@ -0,0 +1,48 @@
+import { isNumber } from "../utils/helpers";
+import { CheckMaxi } from "./check-maxi";
+import { Commands } from "./command";
+import { StoreParameterCommand } from "./store-parameter-command";
+
+export class SetReinvest extends StoreParameterCommand {
+
+    private reinvest: string|undefined
+
+    private usageMessage: string = "/setReinvest 5\nwill result in\nreinvest = 5"
+
+    parseCommandData(): void {
+        if (this.commandData.length === 2) {
+            this.reinvest = this.commandData[1]
+        }
+    }
+
+    validationErrorMessage(): string {
+        return "Input parameter failed validation. Please use following\n" + this.usageMessage
+    }
+
+    validate(): boolean {
+        return isNumber(this.reinvest)
+    }
+
+    successMessage(): string | undefined {
+        return "Your vault-maxis' reinvest is set to " + this.reinvest
+    }
+
+    name(): string {
+        return Commands.SetReinvest
+    }
+
+    description(): string {
+        return "sets given value as reinvest. After changing reinvest it will automatically execute " + Commands.CheckMaxi + " to check if configuration is still valid.\nexample: " + this.usageMessage
+    }
+
+    async doExecution(): Promise<unknown> {
+        if (this.reinvest === undefined) {
+            // Krysh: will never be executed, as validation should fail
```

Comment:
why have it in then? could just say `this.reinvest!` further down, right?

Reply:
still not a fan of using `!`, but yes should be safe here